src/lava/proc/graded/process.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import os
+import numpy as np
+import typing as ty
+from typing import Any, Dict
+from enum import IntEnum, unique
+
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.variable import Var
+from lava.magma.core.process.ports.ports import InPort, OutPort
+
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.magma.core.model.py.ports import PyInPort, PyOutPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.resources import CPU
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+
+
+def loihi2round(vv):
+    """
+    Round values in numpy array the way loihi 2
+    performs rounding/truncation.
+    """
+    return np.fix(vv + (vv > 0) - 0.5).astype('int')
+
+
+class GradedVec(AbstractProcess):
```

Comment:
I think the naming feedback I gave in my other review might have been misconstrued. The issue with these names is they're all just very coded. You should consider that Lava and its libs have a ton of other crap in them, including like, 3 more entire computing paradigms. A name like GradedVec just doesn't give me any insight into what this does or why or how it relates to anything else. This is made more challenging by the fact that your docstring is just restating the expanded name, and your package name is just lava.proc.graded, so that doesn't help either.

Here are some naming proposals that I would find "scrutable", but I encourage you to just think about what a full, properly descriptive name for objects in this system, so that all your processes can be consistent. Maybe that's what you were already aiming for with "Vec", if so, can you just explain that somewhere so users will understand?

lava.proc.vector_algebra.ThresholdVector
lava.proc.vector_symbols.ThresholdVector
lava.proc.vsa.ThresholdNeuron
lava.proc.vector_algebra.GradedThresholdVector
lava.proc.vsa.ThresholdVec  << If you use Vec consistently, and explain it in the docs, e.g. an AbstractVec class.

Less good, but maybe okay:
lava.proc.vector_operations.Threshold
lava.proc.math.Threshold